src/components/alarm/AlarmController.cpp

```diff
@@ -88,20 +111,67 @@ uint32_t AlarmController::SecondsToAlarm() const {
 
 void AlarmController::DisableAlarm() {
   xTimerStop(alarmTimer, 0);
-  state = AlarmState::Not_Set;
+  isAlerting = false;
+  if (alarm.isEnabled) {
+    alarm.isEnabled = false;
+    alarmChanged = true;
+  }
 }
 
 void AlarmController::SetOffAlarmNow() {
-  state = AlarmState::Alerting;
+  isAlerting = true;
   systemTask->PushMessage(System::Messages::SetOffAlarm);
 }
 
 void AlarmController::StopAlerting() {
-  // Alarm state is off unless this is a recurring alarm
-  if (recurrence == RecurType::None) {
-    state = AlarmState::Not_Set;
+  isAlerting = false;
+  // Disable alarm unless it is recurring
+  if (alarm.recurrence == RecurType::None) {
+    alarm.isEnabled = false;
+    alarmChanged = true;
   } else {
     // set next instance
     ScheduleAlarm();
   }
 }
+
+void AlarmController::SetRecurrence(RecurType recurrence) {
+  if (alarm.recurrence != recurrence) {
+    alarm.recurrence = recurrence;
+    alarmChanged = true;
+  }
+}
+
+void AlarmController::Restore() {
```

Comment:
Maybe `LoadAlarmFile` or similar could be more descriptive?

Reading Init
```c++
void AlarmController::Init(System::SystemTask* systemTask) {
  this->systemTask = systemTask;
  alarmTimer = xTimerCreate("Alarm", 1, pdFALSE, this, SetOffAlarm);
  Restore();
  ```
  The meaning of `Restore` isn't immediately clear to me
  If renaming, probably change `Save` too

Reply:
the original function was changed from `LoadAlarmFromFile`/`SaveAlarmToFile` to `Restore`/`Save` because of feedback from JF. But the discussed part was the redundancy of the `Alarm` work in the function. So I think a rename to `LoadSettingsFromFile()`/`SaveSettingsToFile()` should be OK

https://github.com/InfiniTimeOrg/InfiniTime/pull/1367#discussion_r1007262187

Reply:
done!